src/app/shared/log-in/log-in-container/log-in-container.component.ts

```diff
@@ -0,0 +1,109 @@
+import {Component, Injector, OnDestroy, OnInit} from '@angular/core';
+import {Store} from '@ngrx/store';
+import {LogInComponent} from '../password/log-in.component';
+import {DynamicShibbolethComponent} from '../shibboleth/dynamic-shibboleth.component';
+import {getAuthenticationMethods} from '../../../core/auth/selectors';
+import {map} from 'rxjs/operators';
+import {AppState} from '../../../app.reducer';
+import {Observable} from 'rxjs';
+import {DynamicLoginMethod} from './log-in-container.model';
+import {AuthMethodConstants} from '../../../core/auth/models/auth-method.model';
+import {DynamicLdapComponent} from '../ldap/dynamic-ldap.component';
+import {DynamicX509Component} from '../x509/dynamic-x509.component';
+import {DynamicIpComponent} from '../ip/dynamic-ip.component';
+
+@Component({
+  selector: 'ds-log-in-container',
+  templateUrl: './log-in-container.component.html',
+  styleUrls: ['./log-in-container.component.scss'],
+
+})
+export class LogInContainerComponent implements OnDestroy, OnInit {
+
+  public dynamicLoginMethods: Observable<DynamicLoginMethod[]>;
+  /**
+   * Injector to inject a section component with the @Input parameters
+   * @type {Injector}
+   */
+  public objectInjector: Injector;
+  private shibbolethUrl: string;
+
+  /**
+   * @constructor
+   * @param {Store<State>} store
+   * @param {Injector} injector
+   */
+  constructor(
+    private store: Store<AppState>,
+    private injector: Injector
+  ) {
+  }
+
+  /**
+   * Lifecycle hook that is called after data-bound properties of a directive are initialized.
+   * @method ngOnInit
+   */
+  public ngOnInit() {
+
+    this.objectInjector = Injector.create({
+      providers: [
+        {provide: 'shibbolethUrlProvider', useFactory: () => (this.shibbolethUrl), deps: []},
+        // if other authentication methods need further data to work add a provider here e.g
+        // {provide: 'otherDataProvider', useFactory: () => (this.otherData), deps: []},
+      ],
+      parent: this.injector
+    });
+
+    this.dynamicLoginMethods = this.store.select(getAuthenticationMethods).pipe(
+      map(((authMethods) => authMethods.map((authMethod) => {
```

Comment:
I'd avoid to have a fixed switch case, but I'd use a decorator to dinamically add login method.
[Here](https://github.com/DSpace/dspace-angular/blob/master/src/app/submission/sections/sections-decorator.ts) an example and [here](https://github.com/DSpace/dspace-angular/blob/master/src/app/submission/sections/form/section-form.component.ts#L40) how is used

Reply:
Changed it to the approach with decorator and ngComonentOutlet directive